aea/helpers/dialogue/base.py

```diff
@@ -602,13 +616,17 @@ def _additional_rules(self, message: Message) -> bool:
         :param message: the message to be validated
         :return: True if valid, False otherwise.
         """
-        if self.last_message is None:
-            result = True
-        else:
-            target = message.target
+        if self.last_message is not None:
             last_target = self.last_message.target
-            result = target == last_target + 1
-        return result
+            if message.target != last_target + 1:
+                return (
+                    False,
+                    "Invalid target. Expected {}. Found {}.".format(
+                        last_target + 1, message.target
+                    ),
+                )
+
+        return True, "The message passes additional validation."
```

Comment:
This is the opposite of a guard clause; why not do
```
if self.last_message is None:
     return True, "The message passes additional validation."

...
```

Reply:
This is much easier to read and avoids one level of nesting!

Reply:
There would then be two `else` clauses returning `True` + string info, one the opposite of `if self.last_message is not None:` the other the opposite of `if message.target != last_target + 1`

Reply:
yes; just have a few guard clauses, easier to read:
```
if ...
 return
if ...
  return
...
return
```

Reply:
Done